.github/workflows/k8s-build-and-test.yml

```diff
@@ -0,0 +1,43 @@
+# Copyright 2021 Vectorized, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+name: K8S build and test
+on: [push, pull_request]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
```

Comment:
Do we use ubuntu for workflows in general? Are multiple OSes an option?

Reply:
In general we could use strategy matrix os [supported runners](https://docs.github.com/en/actions/reference/specifications-for-github-hosted-runners#supported-runners-and-hardware-resources). My only concern is why? The operator will be running inside the kubernetes cluster which will be Linux machine.

On the other hand `rpk build` uses [darwin and linux](https://github.com/vectorizedio/redpanda/blob/6daa91d55f7f3367e62eb5db56b05af0c37c01f1/.github/workflows/rpk-build.yml#L27) because it's binary distributed on client machines.

P.S. `rpk` should be build for windows environment too CC @0x5d

Reply:
I meant multiple linux distributions. I typically run the operator directly on the host and I use Fedora so I'm hoping it keeps working there. :)

Reply:
Oh, I get your point. So there will be problem in regards to fedora distribution. Github doesn't provide VM Fedora runner. E2E are tight to VM. I can run unit test and integration test inside the fedora container. The kuttl can not be run inside container, because there is problem with the docker network. Kind run in "kind" network, but the runner is in different one. I don't see way to specify or override the network name so that both are in the same.